### PR TITLE
Add org, region resources and  org, vdc, region, supervisor data sources

### PR DESCRIPTION
**Pull request by Didainius**
Changelog contains the list of implemented resources. 
`make testtm` works

### CODE REVIEW

File: website/docs/r/service_account.html.markdown [hunk 1]
```diff
@@ -26,9 +26,9 @@ data "vcd_role" "vapp_author" {
 }
 
 resource "vcd_service_account" "example_service" {
-  org  = "my-org"
-  name = "example"
-  role = data.vcd_role.vapp_author.id
+  org     = "my-org"
+  name    = "example"
+  role_id = data.vcd_role.vapp_author.id
```
Comment (Didainius) [inline website/docs/r/service_account.html.markdown:31]: Patch incorrect field name in docs

File: vcd/datasource_vcd_tm_org.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+				Description: "Number of users in the Org",
+			},
+			"disk_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of disks in the Org",
+			},
+			"can_publish": {
+				Type:     schema.TypeBool,
+				Computed: true,
```
Comment (adambarreiro) [inline vcd/datasource_vcd_tm_org.go:72]: Missing `Description`

File: vcd/datasource_vcd_tm_org.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+			},
+			"directly_managed_org_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of directly managed Orgs",
+			},
+			"is_classic_tenant": {
+				Type:     schema.TypeBool,
+				Computed: true,
```
Comment (adambarreiro) [inline vcd/datasource_vcd_tm_org.go:81]: Missing `Description`

File: vcd/datasource_vcd_tm_region.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+				Computed:    true,
+				Description: "A set supervisor IDs used in this Region",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"storage_policies": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "A set storage policies",
```
Comment (adambarreiro) [inline vcd/datasource_vcd_tm_region.go:71]: Just to double check, these are not IDs, right?
```suggestion
				Description: "A set of storage policies",
```

File: vcd/datasource_vcd_tm_region.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+	vcdClient := meta.(*VCDClient)
+
+	region, err := vcdClient.GetRegionByName(d.Get("name").(string))
+	if err != nil {
+		return diag.Errorf("error retrieving Region: %s", err)
+	}
+
+	err = setRegionData(d, region.Region)
+	if err != nil {
+		return diag.Errorf("error storing data: %s", err)
```
Comment (adambarreiro) [inline vcd/datasource_vcd_tm_region.go:90]:
```suggestion
		return diag.Errorf("error storing Region data: %s", err)
```

File: vcd/datasource_vcd_tm_supervisor.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		},
+	}
+}
+
+func datasourceVcdTmSupervisorRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	s, err := vcdClient.GetSupervisorByName(d.Get("name").(string))
+	if err != nil {
+		return diag.Errorf("error getting Org: %s", err)
```
Comment (adambarreiro) [inline vcd/datasource_vcd_tm_supervisor.go:40]:
```suggestion
		return diag.Errorf("error getting Supervisor: %s", err)
```

File: vcd/datasource_vcd_tm_vdc.go [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+			},
+			"zone_resource_allocations": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Elem:        tmVdcDsZoneResourceAllocation,
+				Description: "A set of Supervisor Zones and their resource allocations",
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
```
Comment (adambarreiro) [inline vcd/datasource_vcd_tm_vdc.go:56]: Missing `Description`

File: vcd/resource_vcd_tm_org.go [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+				Description: "Number of users in the Org",
+			},
+			"disk_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of disks in the Org",
+			},
+			"can_publish": {
+				Type:     schema.TypeBool,
+				Computed: true,
```
Comment (adambarreiro) [inline vcd/resource_vcd_tm_org.go:82]: Missing `Description`

File: vcd/datasource_vcd_tm_region.go [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+			},
+			"is_enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Defines whether the Region is enabled or not",
+			},
+			"nsx_manager_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "NSX-T Manager ID",
```
Comment (adambarreiro) [inline vcd/datasource_vcd_tm_region.go:33]: To have it equal to the resource Description
```suggestion
				Description: "NSX Manager ID",
```

File: vcd/resource_vcd_tm_region.go [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+				Computed:    true,
+				Description: "A set supervisor IDs used in this Region",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"storage_policies": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "A set storage policies",
```
Comment (adambarreiro) [inline vcd/resource_vcd_tm_region.go:81]:
```suggestion
				Description: "A set of storage policies",
```

File: vcd/resource_vcd_tm_region.go [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Status of the region",
+			},
+			"supervisors": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "A set supervisor IDs used in this Region",
```
Comment (adambarreiro) [inline vcd/resource_vcd_tm_region.go:73]:
```suggestion
				Description: "A set of supervisor IDs used in this Region",
```

**Review by adambarreiro — CHANGES_REQUESTED** [review]
First scan (schemas, logs, etc)

File: vcd/resource_vcd_tm_region.go (hunk 11 above)
Comment (Didainius) [reply vcd/resource_vcd_tm_region.go]: Fixed.

File: vcd/resource_vcd_tm_region.go (hunk 10 above)
Comment (Didainius) [reply vcd/resource_vcd_tm_region.go]: Fixed

File: vcd/datasource_vcd_tm_region.go (hunk 9 above)
Comment (Didainius) [reply vcd/datasource_vcd_tm_region.go]: Fixed

File: vcd/resource_vcd_tm_org.go (hunk 8 above)
Comment (Didainius) [reply vcd/resource_vcd_tm_org.go]: Fixed.

File: vcd/datasource_vcd_tm_vdc.go (hunk 7 above)
Comment (Didainius) [reply vcd/datasource_vcd_tm_vdc.go]: fixed

File: vcd/datasource_vcd_tm_supervisor.go (hunk 6 above)
Comment (Didainius) [reply vcd/datasource_vcd_tm_supervisor.go]: Fixed.

File: vcd/datasource_vcd_tm_region.go (hunk 5 above)
Comment (Didainius) [reply vcd/datasource_vcd_tm_region.go]: fixed.

File: vcd/datasource_vcd_tm_region.go (hunk 4 above)
Comment (Didainius) [reply vcd/datasource_vcd_tm_region.go]: fixed.

File: vcd/datasource_vcd_tm_org.go (hunk 3 above)
Comment (Didainius) [reply vcd/datasource_vcd_tm_org.go]: Fixed.

File: vcd/datasource_vcd_tm_org.go (hunk 2 above)
Comment (Didainius) [reply vcd/datasource_vcd_tm_org.go]: Fixed.

File: .changes/v4.0.0/1338-features.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+* **New Resource:** `vcd_tm_org` to manage Organizations using OpenAPI [GH-1338]
+* **New Data Source:** `vcd_tm_org` to read Organizations using OpenAPI [GH-1338]
+* **New Resource:** `vcd_tm_region` to manage Regions [GH-1338]
+* **New Data Source:** `vcd_tm_region` to read Regions [GH-1338]
+* **New Data Source:** `vcd_tm_region` to read VDC using OpenAPI [GH-1338]
```
Comment (adambarreiro) [inline .changes/v4.0.0/1338-features.md:5]:
```suggestion
* **New Data Source:** `vcd_tm_vdc` to read VDC using OpenAPI [GH-1338]
```

**Review by adambarreiro — APPROVED** [review]
Tests passed for me

File: .changes/v4.0.0/1338-features.md (hunk 12 above)
Comment (Didainius) [reply .changes/v4.0.0/1338-features.md]: Fixed.

File: vcd/datasource_vcd_tm_region.go [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+			},
+			"memory_reservation_capacity_mib": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Memory reservation in MiB",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Status of the region",
```
Comment (lvirbalas) [inline vcd/datasource_vcd_tm_region.go:58]: Nit: in some descriptions it's "Region" and here "region". Let's use one way.

**Review by lvirbalas — COMMENTED** [review]
Will wait for docs.

**Comment by Didainius** [comment]
Splitting into pieces.

**Pull request closed without merging**